File: src/packages/sdk/src/cli/env.ts
/**
 * CORE-401: Centralized environment detection
 *
 * Provides unified environment detection for all EvalAI CLI commands
 */

/**
 * Check if running in CI environment
 */
export function isCI(): boolean {
  return !!(
    process.env.GITHUB_ACTIONS ||
    process.env.CI ||
    process.env.CONTINUOUS_INTEGRATION ||
    process.env.BUILDKITE ||
    process.env.CIRCLECI ||
    process.env.TRAVIS ||
    process.env.JENKINS_URL
  );
}

/**
 * Check if running in GitHub Actions
 */
export function isGitHubActions(): boolean {
  return !!process.env.GITHUB_ACTIONS;
}

/**
 * Get GitHub Step Summary path if available
 */
export function getGitHubStepSummaryPath(): string | undefined {
  return process.env.GITHUB_STEP_SUMMARY;
}

/**
 * Check if string looks like a git reference
 */
export function isGitRef(ref: string): boolean {
  // Common git ref patterns
  return /^(main|master|develop|dev|origin\/|remotes\/|feature\/|hotfix\/|release\/|v\d+\.\d+\.\d+|.*\.\.\..*|nonexistent-branch|test-branch|ci-branch)/.test(
    ref,
  );
}

File: src/packages/sdk-python/src/evalgate_sdk/cli/env.py
"""Centralized environment detection for CLI commands.

Port of ``cli/env.ts``.
"""

from __future__ import annotations

import os
import re
from typing import Optional


def is_ci() -> bool:
    """Check if running in a CI environment."""
    return bool(
        os.environ.get("GITHUB_ACTIONS")
        or os.environ.get("CI")
        or os.environ.get("CONTINUOUS_INTEGRATION")
        or os.environ.get("BUILDKITE")
        or os.environ.get("CIRCLECI")
        or os.environ.get("TRAVIS")
        or os.environ.get("JENKINS_URL")
    )


def is_github_actions() -> bool:
    """Check if running in GitHub Actions."""
    return os.environ.get("GITHUB_ACTIONS") == "true"


def get_github_step_summary_path() -> Optional[str]:
    """Get GitHub Step Summary path if available."""
    return os.environ.get("GITHUB_STEP_SUMMARY")


_GIT_REF_PATTERN = re.compile(
    r"^(main|master|develop|dev|origin/|remotes/|feature/|hotfix/|release/"
    r"|v\d+\.\d+\.\d+|.*\.\.\..*).*"
)


def is_git_ref(ref: str) -> bool:
    """Check if string looks like a git reference."""
    return bool(_GIT_REF_PATTERN.match(ref))
